frontend/src/routes/Governance/discovered/grouping.ts

```diff
@@ -0,0 +1,226 @@
+    transformAnnotation: TTransformAnnotation
+  ) => ISourceType
+  createMessage: (
+    data: any[],
+    helmReleases: any[],
+    channels: any[],
+    subscriptions: any[],
+    resolveSource: string,
+    getSourceText: string,
+    getStringLabelValue: string,
+    transformAnnotation: string,
+    parseDiscoveryPolicies: string
+  ) => any[]
+} {
+  const getPolicySource = (
+    policy: any,
+    helmReleases: any[],
+    channels: any[],
+    subscriptions: any[],
+    resolveSource: TResolveSource,
+    getSourceText: TGetSourceText,
+    getStringLabelValue: TGetStringLabelValue,
+    transformAnnotation: TTransformAnnotation
+  ): ISourceType => {
+    const parentPolicy = getStringLabelValue(policy.label, 'policy.open-cluster-management.io/policy')
+    const parentNs = parentPolicy?.split('.')[0] ?? ''
+    const parentName = parentPolicy?.split('.')[1] ?? ''
+
+    if (parentNs && parentName) {
+      return {
+        type: 'Policy',
+        parentNs,
+        parentName,
+      }
+    }
+
+    const isExternal = policy._isExternal
+    let source: string = 'Local'
+    if (isExternal) {
+      const policySource = resolveSource(transformAnnotation(policy.annotations), helmReleases, channels, subscriptions)
+      source = policySource ? getSourceText(policySource, isExternal, (s: string) => s) : 'Managed externally'
+    }
+    // source is string
+    return {
+      type: source,
+      parentNs: '',
+      parentName: '',
+    }
+  }
+
+  enum PolicySeverity {
+    Unknown,
+    Low,
+    Medium,
+    High,
+    Critical,
+  }
+
+  const severityTable = (severity: string): number => {
+    if (!severity) return PolicySeverity.High
+
+    switch (severity.toLowerCase()) {
+      case 'low':
+        return PolicySeverity.Low
+      case 'medium':
+        return PolicySeverity.Medium
+      case 'high':
+        return PolicySeverity.High
+      case 'critical':
+        return PolicySeverity.Critical
+      default:
+        // unknown or null
+        return PolicySeverity.High
+    }
+  }
+
+  const createMessage = (
+    data: any[],
+    helmReleases: any[],
+    channels: any[],
+    subscriptions: any[],
+    resolveSourceStr: string,
+    getSourceTextStr: string,
+    getStringLabelValueStr: string,
+    transformAnnotationStr: string,
+    parseDiscoveryPoliciesStr: string
+  ): any[] => {
+    const resolveSource = eval(resolveSourceStr) as TResolveSource
+    const getSourceText = eval(getSourceTextStr) as TGetSourceText
+    const getStringLabelValue = eval(getStringLabelValueStr) as TGetStringLabelValue
+    const transformAnnotation = eval(transformAnnotationStr) as TTransformAnnotation
+    const parseDiscoveryPolicies = eval(parseDiscoveryPoliciesStr) as TParseDiscoveryPolicies
+
+    const policiesWithSource = (parseDiscoveryPolicies(data) as any[])?.map((policy: any): any => {
+      return {
+        ...policy,
+        remediationAction: policy.remediationAction,
```

Comment:
This isn't included from the `...policy` above this line?

Reply:
Spread `policy` which is from `map((policy: any)`